AAU-P5-Moodle/moodle-1
# Add --- WebSocket server and connectivity to frontend to receive data

# Description
Added a WebSocket server running on port 8001. Established WebSocket connection for when:
* Teachers create quiz rooms, and students can join using the code provided by the teacher.
* Receives data from quiz answers, currently set up to collect data from the question_section.mustache page.

## What is added/changed/removed
WebSocket connection and server.

## Issue/fixes reference

## Testing
No test cases written yet.

Manual testing for creating and joining room
* Create room as teacher
* Join room as student using room code from teacher
* See if users who already joined gets pinged in the websites console

Manual testing for taking quiz
* When taking quiz, check if connection to websocket is established in the websites console
* Check if data gets sent to backend (can be seen in the running terminal)


# Checks
